### DISCUSSION TITLE
Launching forked child processes

### DISCUSSION THREAD

**Discussion by vdawg-git** [cawa-93/vite-electron-builder#808] [Q&A] [answered]
How can I launch a child process with `child_process.fork`?
The `.ts` file needs to get transpiled and launched, while not being part of the bundle. 
How can I achieve that? I thought about adding a new package for that, like `backend` with the others `main`, `renderer` and `preload`.
How could I do this?

**Comment by cawa-93** [comment, marked as answer]
You probably should just copy `main` package. It configured to compile into regular common js module.

**Reply by vdawg-git** [reply]
I realized that this might be a bit overkill. 
Maybe having folder inside `main/src/` called `modules` which does not get bundled but only compiled and put into `dist` as `.cjs` would make it easier.
I am looking through the Rollup docs for that now, but there a ton option. Do you maybe know an easy way to achieve this? :D

**Reply by vdawg-git** [reply]
Actually I think your answer is a nice and simple solution. I will tinker with the results, thank you

**Reply by vdawg-git** [reply]
Okay, I need a bit of help from you. How do I modify `watch.js`, so that the new `backend` will be build upon run and trigger a reload of the app when modified?
Sorry to bother you again :D

**Reply by cawa-93** [reply]
`getWatcher` actually run `vite watch` instance. And there use [`writeBundle` hook](https://rollupjs.org/guide/en/#writebundle) to do something. For example `setupMainPackageWatcher` create vite watch instance and in `writeBundle` do full restart electron app and pass renderer dev URL to app. `setupPreloadPackageWatcher` create vite watch instance and in `writeBundle` send WebSocked message to renderer vite instance for reload web page. In fact this is why we need run `renderer` first and then run preload and main watchers

**Reply by cawa-93** [reply]
Pushed a bit refactoring and docs to watch.js. See https://github.com/cawa-93/vite-electron-builder/commit/634d2d96773c54a3c14a09f3d5ca0f8dce05b10e#diff-7c38a1e3898f31d7b70abe0ed9f6c8f9c8718fad18d5beb636e52c6ad910872b

**Reply by vdawg-git** [reply]
You are the man! Aight, I will update my files and get it to work :D
